src/sentry/search/snuba/backend.py

```diff
@@ -250,89 +245,38 @@ def _query(self, projects, retention_window_start, group_queryset, tags, environ
-                    start=max(
-                        filter(None, [
-                            retention_window_start,
-                            now - timedelta(days=90)
-                        ])
-                    ),
-                    end=now,
-                    groupby=['project_id'],
-                    filter_keys={
-                        'project_id': list(missed_projects),
-                    },
-                    aggregations=[['uniq', 'group_id', 'group_count']],
-                    referrer='search',
-                )
-
-                cache.set_many({
-                    self._get_project_count_cache_key(project_id): count
-                    for project_id, count in missing_counts.items()
-                }, options.get('snuba.search.project-group-count-cache-time'))
-
-                counts_by_projects.update(missing_counts)
-
-            min_candidates = options.get('snuba.search.min-pre-snuba-candidates')
-            max_candidates = options.get('snuba.search.max-pre-snuba-candidates')
-            candidates_percentage = options.get('snuba.search.pre-snuba-candidates-percentage')
-
-            num_candidates = max(
-                min_candidates,
-                min(
-                    max_candidates,
-                    sum(counts_by_projects.values()) * candidates_percentage
-                )
-            )
-        else:
-            num_candidates = options.get('snuba.search.min-pre-snuba-candidates')
-
-        # pre-filter query
-        candidate_ids = None
-        if num_candidates and limit <= num_candidates:
-            candidate_ids = list(
-                group_queryset.values_list('id', flat=True)[:num_candidates + 1]
-            )
-            metrics.timing('snuba.search.num_candidates', len(candidate_ids))
-
-            if not candidate_ids:
-                # no matches could possibly be found from this point on
-                metrics.incr('snuba.search.no_candidates', skip_internal=False)
-                return EMPTY_RESULT
-            elif len(candidate_ids) > num_candidates:
-                # If the pre-filter query didn't include anything to significantly
-                # filter down the number of results (from 'first_release', 'query',
-                # 'status', 'bookmarked_by', 'assigned_to', 'unassigned',
-                # 'subscribed_by', 'active_at_from', or 'active_at_to') then it
-                # might have surpassed the `num_candidates`. In this case,
-                # we *don't* want to pass candidates down to Snuba, and instead we
-                # want Snuba to do all the filtering/sorting it can and *then* apply
-                # this queryset to the results from Snuba, which we call
-                # post-filtering.
-                metrics.incr('snuba.search.too_many_candidates', skip_internal=False)
-                candidate_ids = None
+        # Here we check if all the django filters reduce the set of groups down
```

Comment:
This whole part of the diff is just removing the optimization decision based on how many django results compared to how many overall groups there are, and replacing it with just:
```
if django_results <= max-pre-snuba-candidates:
    # pass them to snuba
```